lib/Command/IndexPageContentCommand.php

```diff
@@ -0,0 +1,170 @@
+    ) {
+        parent::__construct($this::$defaultName);
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Index content related through layouts')
+            ->addOption(
+                'content-ids',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Comma separated list of content id\'s of content to index.',
+            );
+    }
+
+    /**
+     * @throws NotFoundException
+     * @throws InvalidArgumentException
+     * @throws UnauthorizedException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $contentIds = $input->getOption('content-ids');
+        if ($contentIds !== null) {
+            $contentIds = explode(',', $contentIds);
+
+            $totalCount = count($contentIds);
+            $output->writeln("Number of objects to index: {$totalCount}");
+
+            $progressBar = new ProgressBar($output, $totalCount);
+            $progressBar->start();
+            foreach ($contentIds as $contentId) {
+                $content = $this->contentService->loadContent((int) $contentId);
+                $this->indexContentWithLocations($content);
+                $progressBar->advance();
+            }
+        } else {
+            $query = new Query();
+            $offset = 0;
+            $limit = 50;
+            $query->query = new Criterion\ContentTypeIdentifier($this->allowedContentTypes);
+            $totalCount = $this->getTotalCount($query);
+            $progressBar = new ProgressBar($output, $totalCount);
+
+            if ($totalCount <= 0) {
+                $output->writeln('No content found to index, exiting.');
+
+                return Command::SUCCESS;
+            }
+
+            $output->writeln('Found ' . $totalCount . ' content objects...');
+            $output->writeln('');
+
+            $progressBar->start($totalCount);
+
+            while ($offset < $totalCount) {
+                $chunk = $this->getChunk($query, $limit, $offset);
+
+                $this->processChunk($chunk, $output, $progressBar);
+
+                $offset += $limit;
+            }
+
+            $progressBar->finish();
+
+            $output->writeln('');
+            $output->writeln('');
+            $output->writeln('Finished.');
+        }
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    private function getTotalCount(Query $query): int
+    {
+        $filter = new Filter();
+        $filter
+            ->withCriterion(
+                new Query\Criterion\ContentTypeIdentifier($this->allowedContentTypes)
+            )
+            ->withLimit(0)
+            ->withOffset(0)
+        ;
+
+        return $this->contentService->find($filter)->getTotalCount() ?? 0;
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    private function getChunk(Query $query, int $limit, int $offset): ContentList
```

Comment:
Argument `$query` is not used.

Reply:
resolved in https://github.com/netgen/ibexa-search-extra/commit/28c2082565f4b56880ebae582f4b8de43fedcae3